File: cpp/temp/Label.cpp
#include "Label.h"
#include "tiger.h"
#include "debug.h"

Label::Label()
	: name(format("L%d", count++))
{
}

Label::Label(const std::string &name)
	: name(name) 
{
}

Label::~Label()
{
}

const std::string & 
Label::toString() const
{
	return name;
}

int
Label::count;

File: scala/src/main/scala/tiger/common/Label.scala
package tiger.common

import tiger.tree._

object Label {
	var count:Int = 0
/*
	def ::(rhs:Label):Pair[Label] = {
		(this, rhs)
	}
*/
}

class Label(sym:Symbol) {
	def this() = {
		this(Symbol("L" + Label.count))
		Label.count += 1
	}
	override def toString():String = sym.name
}

